lib/cli-engine/config-array-factory.js

```diff
@@ -722,7 +722,12 @@ class ConfigArrayFactory {
          *
          * Refer https://github.com/eslint/eslint/issues/12592
          */
-        const clonedRulesConfig = rules && JSON.parse(JSON.stringify((rules)));
+        const clonedRulesConfig = rules && JSON.parse(
+            JSON.stringify(
+                rules,
+                (key, value) => (value === Infinity ? Number.MAX_SAFE_INTEGER : value)
```

Comment:
```suggestion
                (key, value) => (value === Infinity ? Number.MAX_SAFE_INTEGER : (value === -Infinity : -Number.MAX_SAFE_INTEGER : value))
```
just in case?

Reply:
I'd prefer not to add any further exceptions here. This fix is to carve out a narrow hole for `eslint-config-airbnb` to fit through without allowing other exceptions through.

Reply:
I suppose we can just wait for bug reports from enterprise users, which upgrade eslint much more slowly than open source users. Either way JSON-cloning is a breaking change, so if you'd prefer to gamble that it won't come up, that's fine ¯\\\_(ツ)_/¯

Reply:
Has anybody considered using something like https://github.com/codeandcats/fast-clone instead of JSON cloning?

Reply:
(actually: https://github.com/davidmarkclements/rfdc)

Reply:
@eaviles See https://github.com/eslint/eslint/pull/13034 for the original (long 😆) discussion.

Reply:
Oh wow… thanks for pointing me to this @kaicataldo

Reply:
Just as a reminder (and for posterity): ESLint has never formally supported JS special values in rule configs, which originated as JSON configs and are validated with JSON Schema. A validation bug let people use Infinity and we only learned of that after implementing this change. We don’t want to open the door to more people using special values in the meantime, which is why we are not reverting the original change.

If other bugs are filed, we will handle them on a case-by-case basis. Keep in mind that a large number of users use JSON or YAML config files, and this issue only affects JS config files.